aospy/data_loader.py

```diff
@@ -439,12 +481,20 @@ def __init__(self, template=None, data_direc=None, data_dur=None,
                 self.preprocess_func = preprocess_func
             else:
                 self.preprocess_func = template.preprocess_func
+            if upcast_float32 is not None:
+                self.upcast_float32 = upcast_float32
+            else:
+                self.upcast_float32 = template.upcast_float32
         else:
             self.data_direc = data_direc
             self.data_dur = data_dur
             self.data_start_date = data_start_date
             self.data_end_date = data_end_date
             self.preprocess_func = preprocess_func
+            if upcast_float32 is None:
```

Comment:
In the case that a template `GFDLDataLoader` is not provided, we translate the default argument of `None` to mean `True`.